lib/compiler/passes/generate-bytecode.js

```diff
@@ -564,13 +585,15 @@ function generateBytecode(ast) {
 
     literal(node) {
       if (node.value.length > 0) {
-        const stringIndex = addConst("\""
+        const match = node.match | 0;
+        const needConst = match === 0 || (match > 0 && !node.ignoreCase);
+        const stringIndex = needConst ? addConst("\""
           + js.stringEscape(
               node.ignoreCase ? node.value.toLowerCase() : node.value
             )
           + "\""
-        );
-        const expectedIndex = addConst(
+        ) : null;
+        const expectedIndex = match > 0 ? null : addConst(
```

Comment:
why is match being compared to zero when it's also being bitwise manipulated everywhere?

Reply:
If you look more carefully, you will see that bitwise operators is used to get the value of the `match` variable. This doesn't magically calculate any comparing results for us

Reply:
> If you look more carefully

People shouldn't have to look more carefully to understand simple code.  This is an excuse.

&nbsp;

> you will see that bitwise operators is used to get the value of the match variable.

This doesn't make any sense.  Bitwise operators are for performing bitwise operations, not getting values of variables.

Reply:
> People shouldn't have to look more carefully to understand simple code. This is an excuse.

All that their need to look, it is to read an introduction comment that coming soon